### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by littlebtc**
Compare with upstream is recommended.
https://github.com/Miserlou/Zappa/compare/master...GeoThings:merge-upstream

Upstream changes:

* Support Python 3.7.
* Native ALB and multi-value supports.

I will try to find a way to test it orz

### CODE REVIEW

File: zappa/wsgi.py [hunk 1]
```diff
@@ -109,13 +119,12 @@ def create_wsgi_request(event_info,
             remote_addr = '127.0.0.1'
 
         environ = {
-            'PATH_INFO': get_wsgi_string(path) if should_transform_to_iso_8859_1 else path,
```
Comment (ambientlight) [inline zappa/wsgi.py:112]: this might break the decoding of non-ascii query string parameters

**Review by ambientlight — CHANGES_REQUESTED** [review]
We have made a few tweaks, I suggest you to go through the commit history of our zappa fork and make sure the fixes we were making are still working as expected.
like you can make a checkbox of our fixes:

- [ ] multi-keep alive originated from https://github.com/Miserlou/Zappa/pull/861 with additional configurable sleep times and MAX_POOL_CONNECTIONS settings available to set on botto3 clients.
- [ ] fix decoding of non-ascii query parameters in django in https://github.com/Miserlou/Zappa/issues/1199
- [ ] multiple query parameters support
- [ ] ALB support with ALB arn specified in zappa settings and ALB lambda resource-based permission requested on updates if needed
- [ ] use thread for run zappa asynchronous @async_task in non-lambda environment
- [ ] _clean_policy should not flush all resource based permissions

File: zappa/wsgi.py (hunk 1 above)
Comment (littlebtc) [reply zappa/wsgi.py:112]: In my experiment it seems the problem was not here, need more investigation so I reverted it.

I will play around with it after we are able to build a version on it.

**Comment by littlebtc** [comment]
As ALB approach seems different and hard to deal with, we may leave this later until we have time to test it 😢

**Comment by ambientlight** [comment]
@littlebtc: ALB will be fine, except that zappa deploy/undeploy will create / remove ALB. 
The difference to our ALB support is that we create a ALB with lambda target group manually and zappa only adds the permissions for ALB to invoke lambda. 
We might need to look into https://github.com/Miserlou/Zappa/pull/1833 for a pluggability with existing ALB which will be easier to integrate with our current setup.

So let's just leave this for now and just deal with python 3.7 support.

**Pull request closed without merging**